public/app/plugins/datasource/influxdb/datasource.ts

```diff
@@ -47,6 +47,9 @@ export function InfluxDatasource(instanceSettings, $q, backendSrv, templateSrv)
     // replace templated variables
     allQueries = templateSrv.replace(allQueries, options.scopedVars);
 
+    // replace database name
+    allQueries = allQueries.replace(/\$database/g, this.database);
```

Comment:
I'll check the documentation of `InfluxDB` and see if I can delete it and fix the pr.